### PR TITLE
[6.x] Add -filled- ability to the support Arr class

### PR DESCRIPTION
**Pull request by gocanto**
This `PR` introduces a new method `filled` for the current `\Illuminate\Support\Arr` class.

This method works as you might expect allowing `dot` notation for nested checks.

Here is the method signature & [implementation](https://github.com/laravel/framework/compare/6.x...gocanto:6.x#diff-3827e5712dc6ff2e21499fcd308a18a1R352): 

```php
public static function filled($array, $keys)
{
   //code
}
```

### The reason behind it

As you might know by now, we currently have two methods to check whether we have a given key(s) within a given array. This works fine, but it does not provide the ability to check whether we have a value for a given key. Hence, you will be able to have this behavior if this `PR` is merged.

### Example

```php
$data = [
            'one' => 1,
            'two' => '',
            'three' => null,
            'foo' => [
                'one' => 1,
                'two' => '',
                'three' => null,
            ],
        ];


if (Arr::filled($data, 'one')) {
//good to go!
} 
```

You will be able to see more uses case [here](https://github.com/laravel/framework/compare/6.x...gocanto:6.x#diff-d14ddb2972fa3aa22bf0f479ae091d60R819)

### Nice to have
If this PR is approved, we could also add an `empty` method. So, we have a positive way to ask in given conditions. 

----
Cheers,

### CODE REVIEW

**Comment by foremtehan** [comment]
I was looking for something like that hope it'll merge

https://laracasts.com/discuss/channels/laravel/check-if-specific-keys-in-collection-has-value-or-not

File: src/Illuminate/Support/Arr.php [hunk 1]
```diff
@@ -342,6 +342,30 @@ public static function has($array, $keys)
...
+    public static function filled($array, $keys)
+    {
+        if (! static::has($array, $keys)) {
+            return false;
+        }
+
+        foreach ((array) $keys as $key) {
+            $needle = static::get($array, $key);
+
+            if (empty($needle)) {
```
Comment (jmarcher) [inline src/Illuminate/Support/Arr.php]: You should use the `blank()` helper here.
The idea of something "filled" could be a bit confusing if we use the `empty()` function as `empty(collect())` returns `false` and `empty([])` returns `true` but both are empty "lists"

**Review by jmarcher — CHANGES_REQUESTED** [review]

File: src/Illuminate/Support/Arr.php (hunk 1 above)
Comment (derekmd) [reply src/Illuminate/Support/Arr.php]: Some more use cases:

```php
empty(0);
// true

empty('0');
// true

empty(false);
// true
```

According to Laravel's `Request::filled()` and the `filled()` global helper, these three examples are considered "present" values, not blank.

**Comment by gocanto** [comment]
Thanks for the feedback guys. I would be able to adjust the `condition` however it's needed. But, I would wait for the `Laravel` final input for me to do so. 

From my point of view, `empty` is the right way to go. But, I am keen to update this method to abide by the `framework` behavior. 

Let's see. 👍

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Pull request closed without merging**